### PR TITLE
When op=list, return golang blank fieldsrather than default when nil

### PR DESCRIPTION
**Pull request by maxkorp**
This way fields in code are represented as their golang default, rather than the default annotation, which is intended for creation.

### CODE REVIEW

File: parse/builder/builder.go [hunk 1]
```diff
@@ -385,6 +390,40 @@ func ConvertSimple(fieldType string, value interface{}, op Operation) (interface
 	return nil, ErrComplexType
 }
 
+func ConvertSimpleNil(fieldType string) (interface{}, error) {
```
Comment (cjellick) [inline parse/builder/builder.go]: Not used outside this package. should not be capitalized

File: parse/builder/builder.go [hunk 2]
```diff
@@ -146,7 +147,11 @@ func (b *Builder) checkDefaultAndRequired(schema *types.Schema, input map[string
 		if op.IsList() && fieldMatchesOp(field, List) && definition.IsReferenceType(field.Type) && !hasKey {
 			result[fieldName] = nil
 		} else if op.IsList() && fieldMatchesOp(field, List) && !hasKey && field.Default != nil {
-			result[fieldName] = field.Default
+			blank, err := ConvertSimpleNil(field.Type)
```
Comment (cjellick) [inline parse/builder/builder.go]: I believe the requirement was that we only give add the key with its zero value if the field is no nullable. right?

File: parse/builder/builder.go [hunk 3]
```diff
@@ -385,6 +397,40 @@ func ConvertSimple(fieldType string, value interface{}, op Operation) (interface
 	return nil, ErrComplexType
 }
 
+func convertSimpleNil(fieldType string) interface{} {
+	switch fieldType {
+	case "json":
+		return "{}"
```
Comment (ibuildthecloud) [inline parse/builder/builder.go]: This should be `map[string]interface{}{}`

File: parse/builder/builder.go [hunk 4]
```diff
@@ -385,6 +397,40 @@ func ConvertSimple(fieldType string, value interface{}, op Operation) (interface
 	return nil, ErrComplexType
 }
 
+func convertSimpleNil(fieldType string) interface{} {
```
Comment (ibuildthecloud) [inline parse/builder/builder.go]: can this be called something related to "empty value" like `getEmptyValue()`

**Comment by ibuildthecloud** [comment]
https://github.com/rancher/norman/pull/223 adds float support and I just merged it, so this PR needs to be updated for floats

**Pull request closed without merging**